### ISSUE TITLE
OpenCV 4.0.1 OSX build framework fails on xcodebuild

### ISSUE THREAD

**Issue by atrbx5** [opencv/opencv#13759]
##### System information (version)

- OpenCV => 4.0.1
- Operating System / Platform => Mojave 10.14.2
- Compiler => CMake 3.13.3
Mac mini (Late 2014) cpu 2.6 GHz Intel Core i5

##### Detailed description

When I try to create framework for OSx I have this error:

```
** BUILD FAILED **

The following build commands failed:
	CompileC platforms/osx/res/build/build-x86_64-macosx/modules/world/OpenCV.build/Release/opencv_world.build/Objects-normal/x86_64/stereobm.o modules/calib3d/src/stereobm.cpp normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
============================================================
ERROR: Command '['xcodebuild', 'MACOSX_DEPLOYMENT_TARGET=10.9', 'ARCHS=x86_64', '-sdk', 'macosx', '-configuration', 'Release', '-parallelizeTargets', '-jobs', '4', '-target', 'ALL_BUILD', 'build']' returned non-zero exit status 65
============================================================
Traceback (most recent call last):
  File "/Users/me/Downloads/opencv-master/platforms/ios/build_framework.py", line 117, in build
    self._build(outdir)
  File "/Users/me/Downloads/opencv-master/platforms/ios/build_framework.py", line 109, in _build
    self.buildOne(t[0], t[1], mainBD, cmake_flags)
  File "/Users/me/Downloads/opencv-master/platforms/ios/build_framework.py", line 205, in buildOne
    execute(buildcmd + ["-target", "ALL_BUILD", "build"], cwd = builddir)
  File "/Users/me/Downloads/opencv-master/platforms/ios/build_framework.py", line 39, in execute
    retcode = check_call(cmd, cwd = cwd)
  File "/usr/local/Cellar/python@2/2.7.15_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 190, in check_call
    raise CalledProcessError(retcode, cmd)
CalledProcessError: Command '['xcodebuild', 'MACOSX_DEPLOYMENT_TARGET=10.9', 'ARCHS=x86_64', '-sdk', 'macosx', '-configuration', 'Release', '-parallelizeTargets', '-jobs', '4', '-target', 'ALL_BUILD', 'build']' returned non-zero exit status 65
```

I"ve also tried MACOSX_DEPLOYMENT_TARGET = 10.14
but got another error:

```
ERROR: Command '['libtool', '-static', '-o', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/lib/Release/libopencv_merged.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/libopencv_world.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/liblibtiff.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/liblibpng.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/liblibwebp.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/libquirc.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/libippiw.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/liblibprotobuf.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/libIlmImf.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/libippicv.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/libade.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/libzlib.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/liblibjasper.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/liblibjpeg-turbo.a']' returned non-zero exit status 1
============================================================
Traceback (most recent call last):
  File "/Users/me/Downloads/opencv-master/platforms/ios/build_framework.py", line 117, in build
    self._build(outdir)
  File "/Users/me/Downloads/opencv-master/platforms/ios/build_framework.py", line 112, in _build
    self.mergeLibs(mainBD)
  File "/Users/me/Downloads/opencv-master/platforms/ios/build_framework.py", line 213, in mergeLibs
    execute(["libtool", "-static", "-o", res] + libs + libs3)
  File "/Users/me/Downloads/opencv-master/platforms/ios/build_framework.py", line 39, in execute
    retcode = check_call(cmd, cwd = cwd)
  File "/usr/local/Cellar/python@2/2.7.15_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 190, in check_call
    raise CalledProcessError(retcode, cmd)
CalledProcessError: Command '['libtool', '-static', '-o', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/lib/Release/libopencv_merged.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/libopencv_world.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/liblibtiff.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/liblibpng.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/liblibwebp.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/libquirc.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/libippiw.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/liblibprotobuf.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/libIlmImf.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/libippicv.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/libade.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/libzlib.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/liblibjasper.a', '/Users/me/Downloads/opencv-master/platforms/osx/res/build/build-x86_64-macosx/install/lib/3rdparty/liblibjpeg-turbo.a']' returned non-zero exit status 1

```

I've tried master branch and also get same error but for 10.13 it get finally compiled
and when I dropped framework into osx project I got errors:

```
Undefined symbols for architecture x86_64:
  "_cblas_cgemm", referenced from:
      lapack_gemm32fc(float const*, unsigned long, float const*, unsigned long, float, float const*, unsigned long, float, float*, unsigned long, int, int, int, int) in opencv2(hal_internal.o)
      lapack_gemm64fc(double const*, unsigned long, double const*, unsigned long, double, double const*, unsigned long, double, double*, unsigned long, int, int, int, int) in opencv2(hal_internal.o)
  "_cblas_dgemm", referenced from:
      lapack_gemm64f(double const*, unsigned long, double const*, unsigned long, double, double const*, unsigned long, double, double*, unsigned long, int, int, int, int) in opencv2(hal_internal.o)
  "_cblas_sgemm", referenced from:
      lapack_gemm32f(float const*, unsigned long, float const*, unsigned long, float, float const*, unsigned long, float, float*, unsigned long, int, int, int, int) in opencv2(hal_internal.o)
      lapack_gemm64f(double const*, unsigned long, double const*, unsigned long, double, double const*, unsigned long, double, double*, unsigned long, int, int, int, int) in opencv2(hal_internal.o)
  "_cblas_zgemm", referenced from:
      lapack_gemm64fc(double const*, unsigned long, double const*, unsigned long, double, double const*, unsigned long, double, double*, unsigned long, int, int, int, int) in opencv2(hal_internal.o)
  "_clBuildProgram", referenced from:
      _clBuildProgram_pfn in opencv2(opencl_core.o)
```

##### Steps to reproduce
cd osx
python build_framework.py res

**Comment by alalek** [comment]
The first error message is incomplete. Reason of the error should be above the cut.
The second is the same.


> _cblas_cgemm

These symbols should be available in the framework "Accelerate".

> _clBuildProgram

Framework "OpenCL".

**Comment by atrbx5** [comment]
Thanks, missed frameworks helps compile program.
But is this is bug in the opencv build_framework.py?
And they need to point  MACOSX_DEPLOYMENT_TARGET to 10.13?

**Comment by alalek** [comment]
It is possible to build OpenCV without requirement of these frameworks (optional dependencies).
So framework's plist file configuration is not easy.

> MACOSX_DEPLOYMENT_TARGET

[Apple says](https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/cross_development/Configuring/configuring.html):
> This identifies the earliest OS version on which your software can run.

It is hard to say when we should bump this version. It usually depends on Xcode version installed on developers machines.

iOS framework build script has this configuration for `IPHONEOS_DEPLOYMENT_TARGET`: #12921

**Comment by atrbx5** [comment]
Here error for target 10.9

```
In file included from /Users/me/Downloads/opencv-master/modules/calib3d/src/stereobm.cpp:52:
In file included from /Users/me/Downloads/opencv-master/modules/core/include/opencv2/core/hal/intrin.hpp:149:
/Users/me/Downloads/opencv-master/modules/core/include/opencv2/core/hal/intrin_sse_em.hpp:103:1: error: 
      always_inline function '_mm_cvtepu8_epi16' requires target feature
      'sse4.1', but would be inlined into function '_v128_cvtepu8_epi16' that is
      compiled without support for 'sse4.1'
OPENCV_HAL_SSE_WRAP_1(cvtepu8_epi16, __m128i)
^
/Users/me/Downloads/opencv-master/modules/core/include/opencv2/core/hal/intrin_sse_em.hpp:17:14: note: 
      expanded from macro 'OPENCV_HAL_SSE_WRAP_1'
    { return _mm_##fun(a); }
             ^
<scratch space>:95:1: note: expanded from here
_mm_cvtepu8_epi16
^
In file included from /Users/me/Downloads/opencv-master/modules/calib3d/src/stereobm.cpp:52:
In file included from /Users/me/Downloads/opencv-master/modules/core/include/opencv2/core/hal/intrin.hpp:149:
/Users/me/Downloads/opencv-master/modules/core/include/opencv2/core/hal/intrin_sse_em.hpp:104:1: error: 
      always_inline function '_mm_cvtepi8_epi16' requires target feature
      'sse4.1', but would be inlined into function '_v128_cvtepi8_epi16' that is
      compiled without support for 'sse4.1'
OPENCV_HAL_SSE_WRAP_1(cvtepi8_epi16, __m128i)
^
/Users/me/Downloads/opencv-master/modules/core/include/opencv2/core/hal/intrin_sse_em.hpp:17:14: note: 
      expanded from macro 'OPENCV_HAL_SSE_WRAP_1'
    { return _mm_##fun(a); }
             ^
<scratch space>:97:1: note: expanded from here
_mm_cvtepi8_epi16
^
In file included from /Users/me/Downloads/opencv-master/modules/calib3d/src/stereobm.cpp:52:
In file included from /Users/me/Downloads/opencv-master/modules/core/include/opencv2/core/hal/intrin.hpp:149:
/Users/me/Downloads/opencv-master/modules/core/include/opencv2/core/hal/intrin_sse_em.hpp:108:1: error: 
      always_inline function '_mm_cvtepi16_epi32' requires target feature
      'sse4.1', but would be inlined into function '_v128_cvtepi16_epi32' that
      is compiled without support for 'sse4.1'
OPENCV_HAL_SSE_WRAP_1(cvtepi16_epi32, __m128i)
^
/Users/me/Downloads/opencv-master/modules/core/include/opencv2/core/hal/intrin_sse_em.hpp:17:14: note: 
      expanded from macro 'OPENCV_HAL_SSE_WRAP_1'
    { return _mm_##fun(a); }
             ^
<scratch space>:105:1: note: expanded from here
_mm_cvtepi16_epi32
^
In file included from /Users/me/Downloads/opencv-master/modules/calib3d/src/stereobm.cpp:52:
In file included from /Users/me/Downloads/opencv-master/modules/core/include/opencv2/core/hal/intrin.hpp:149:
/Users/me/Downloads/opencv-master/modules/core/include/opencv2/core/hal/intrin_sse_em.hpp:107:1: error: 
      always_inline function '_mm_cvtepu16_epi32' requires target feature
      'sse4.1', but would be inlined into function '_v128_cvtepu16_epi32' that
      is compiled without support for 'sse4.1'
OPENCV_HAL_SSE_WRAP_1(cvtepu16_epi32, __m128i)
^
/Users/me/Downloads/opencv-master/modules/core/include/opencv2/core/hal/intrin_sse_em.hpp:17:14: note: 
      expanded from macro 'OPENCV_HAL_SSE_WRAP_1'
    { return _mm_##fun(a); }
             ^
<scratch space>:103:1: note: expanded from here
_mm_cvtepu16_epi32
^
In file included from /Users/me/Downloads/opencv-master/modules/calib3d/src/stereobm.cpp:52:
In file included from /Users/me/Downloads/opencv-master/modules/core/include/opencv2/core/hal/intrin.hpp:150:
/Users/me/Downloads/opencv-master/modules/core/include/opencv2/core/hal/intrin_sse.hpp:942:22: error: 
      always_inline function '_mm_min_epi32' requires target feature 'sse4.1',
      but would be inlined into function 'v_min' that is compiled without
      support for 'sse4.1'
    return v_int32x4(_mm_min_epi32(a.val, b.val));
                     ^
/Users/me/Downloads/opencv-master/modules/core/include/opencv2/core/hal/intrin_sse.hpp:1582:1: error: 
      always_inline function '_mm_blendv_ps' requires target feature 'sse4.1',
      but would be inlined into function 'v_select' that is compiled without
      support for 'sse4.1'
OPENCV_HAL_IMPL_SSE_SELECT(v_int32x4, _mm_castps_si128, _mm_castsi128_ps, ps)
^
/Users/me/Downloads/opencv-master/modules/core/include/opencv2/core/hal/intrin_sse.hpp:1574:28: note: 
      expanded from macro 'OPENCV_HAL_IMPL_SSE_SELECT'
    return _Tpvec(cast_ret(_mm_blendv_##suffix(cast(b.val), cast(a.val)...
                           ^
<scratch space>:15:1: note: expanded from here
_mm_blendv_ps
^
6 errors generated.
```

**Comment by SoylentGraham** [comment]
It wasn't really obvious from this thread what the solution is...
I just had the same errors on 
- High Sierra/10.13.6 
- imac 2011
- current master(7a2dca)
- Xcode 10.1 (10B61)

in `platforms/osx/build_framework.py` I increased `MACOSX_DEPLOYMENT_TARGET` from `10.9` to `10.12` (arbitrarily trying something above 10.9, I'm not targeting that specifically)
and it built successfully. (Currently untested, but it spat out `opencv2.framework` at least :)

**Comment by SoylentGraham** [comment]
Just came across this again (forgot I posted this, just found my own solution).

For clarity, `10.10`, `10.11` don't work. (Tested a few more cases)

`10.12` does (as of current head f6c573880e57e78e0c07d8832ee6015f2337a019)

**Comment by komakai** [comment]
Fixed in #15350
